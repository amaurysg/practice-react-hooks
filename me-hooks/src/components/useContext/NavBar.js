import React from "react";
import { Link, NavLink } from "react-router-dom";

const NavBar = () => {
  return (
    <nav className="navbar navbar-expand-lg navbar-light bg-light">
      <div className="container-md">
        <NavLink exact to="/" className="navbar-brand active">
          Home
        </NavLink>
        <NavLink exact to="/about" className="navbar-brand">
          About
        </NavLink>
        <NavLink exact to="/login" className="navbar-brand">
          Login
        </NavLink>
      </div>
    </nav>
  );
};

export default NavBar;
