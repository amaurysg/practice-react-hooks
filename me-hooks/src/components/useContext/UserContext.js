import { createContext } from "react";

//1.Here created instance to userContext
export const UserContext = createContext(null);
// const UserContext = () => {

//   return (
//     <div>

//     </div>
//   )
// }
