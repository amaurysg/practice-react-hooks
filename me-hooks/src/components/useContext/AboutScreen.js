import React, { useContext } from "react";
import { UserContext } from "./UserContext";

const AboutScreen = () => {
  const { user, setUser } = useContext(UserContext);

  const handleClick = () => {
    setUser({});
  };

  return (
    <div>
      <h1>About</h1>
      <hr />
      <pre>{JSON.stringify(user, null, 3)}</pre>
      <button className="btn btn-outline-primary" onClick={handleClick}>
        Reset
      </button>
    </div>
  );
};

export default AboutScreen;
