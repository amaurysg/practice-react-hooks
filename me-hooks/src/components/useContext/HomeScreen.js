import React, { useContext } from "react";
import { UserContext } from "./UserContext";

const HomeScreen = () => {
  //1. Here called UserContext from other file.
  const { user, setUser } = useContext(UserContext);

  console.log("user:", user);

  return (
    <div>
      <h1>Home Screen</h1>
      <pre>{JSON.stringify(user, null, 3)}</pre>
    </div>
  );
};

export default HomeScreen;
