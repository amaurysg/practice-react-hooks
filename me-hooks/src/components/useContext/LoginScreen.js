import React, { useContext, useState } from "react";
import { UserContext } from "./UserContext";

const LoginScreen = () => {
  //1.Obtener referencia
  const { setUser } = useContext(UserContext);

  //2. setUser

  return (
    <div>
      <h1>Login</h1>
      <hr />
      <button
        className="btn btn-primary"
        onClick={() => {
          setUser({
            id: 123,
            fname: "Amaury",
          });
        }}
      >
        Log in ðŸ˜„
      </button>
    </div>
  );
};

export default LoginScreen;
