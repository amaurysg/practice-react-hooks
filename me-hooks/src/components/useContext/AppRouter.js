import React from "react";
import {
  BrowserRouter as Router,
  Switch,
  Route,
  Link,
  Redirect,
} from "react-router-dom";
import NavBar from "./NavBar";
import AboutScreen from "./AboutScreen";
import HomeScreen from "./HomeScreen";
import LoginScreen from "./LoginScreen";

const AppRouter = () => {
  return (
    <Router>
      <div>
        <NavBar />
        <Switch>
          <Route exact path="/about" component={AboutScreen} />
          <Route exact path="/login" component={LoginScreen} />
          <Route exact path="/" component={HomeScreen} />
          <Redirect to="/" />
        </Switch>
      </div>
    </Router>
  );
};

export default AppRouter;
