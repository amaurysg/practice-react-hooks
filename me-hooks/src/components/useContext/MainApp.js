import React, { useState } from "react";
import AppRouter from "./AppRouter";
import { UserContext } from "./UserContext";

const MainApp = () => {
  //1.Initial obj empty
  const [user, setUser] = useState({});

  return (
    //2.Here envolved whole the appRouter with component UserContext.Provider
    //3. Share const {user} to all components
    <UserContext.Provider
      value={{
        user,
        setUser,
      }}
    >
      <AppRouter />
    </UserContext.Provider>
  );
};

export default MainApp;
