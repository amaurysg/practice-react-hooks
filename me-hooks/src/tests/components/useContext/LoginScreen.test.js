import { mount } from "enzyme";
import LoginScreen from "../../../components/useContext/LoginScreen";
import { UserContext } from "../../../components/useContext/UserContext";

describe("Test en LoginScreen", () => {
  const setUser = jest.fn();
  //wrapper
  const wrapper = mount(
    <UserContext.Provider value={{ setUser: setUser }}>
      <LoginScreen />
    </UserContext.Provider>
  );

  test("Debe renderizar", () => {
    //snapshot
    expect(wrapper).toMatchSnapshot();
  });
  test("Debe ejecutar el setUser con el arg esperado", () => {
    wrapper.find("button").prop("onClick")();
    expect(setUser).toHaveBeenCalledWith({
      id: 123,
      fname: "Amaury",
    });
  });
});
