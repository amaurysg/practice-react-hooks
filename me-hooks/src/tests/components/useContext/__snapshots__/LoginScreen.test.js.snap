// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test en LoginScreen Debe renderizar 1`] = `
<div>
  <h1>
    Login
  </h1>
  <hr />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Log in ðŸ˜„
  </button>
</div>
`;
